opendbc/car/hyundai/tests/test_hyundai.py

```diff
@@ -44,17 +45,20 @@
 class TestHyundaiFingerprint:
   def test_feature_detection(self):
     # HDA2
-    for has_adas in (True, False):
-      car_fw = [CarParams.CarFw(ecu=Ecu.adas if has_adas else Ecu.fwdCamera)]
-      CP = CarInterface.get_params(CAR.KIA_EV6, gen_empty_fingerprint(), car_fw, False, False)
-      assert bool(CP.flags & HyundaiFlags.CANFD_HDA2) == has_adas
+    for hda2 in (True, False):
+      fingerprint = gen_empty_fingerprint()
+      if hda2:
+        cam_can = CanBus(None, False, fingerprint).CAM
+        fingerprint[cam_can] = [0x50, 0x110]  # HDA2 steering messages
+      CP = CarInterface.get_params(CAR.KIA_EV6, fingerprint, [], False, False)
+      assert bool(CP.flags & HyundaiFlags.CANFD_HDA2) == hda2
```

Comment:
I like that you added a test! Can't do this yet, but for the first car we merge that uses the new CAN fallback, let's duplicate the EV6 check with that car, as an example without the ADAS ECU.

Reply:
I like this idea. Will implement.